File: Railway.UI/src/app/models/procedures/routeSearch/routeStopTicketDto.ts
import { RouteStopTicketDetailDto } from "./routeStopTicketDetailDto";

export class RouteStopTicketDto {
    id?: number;
    routeId?: number;

    routeFromLocality?: string;
    fromRouteStopId?: number;

    routeToLocality?: string;
    toRouteStopId?: number;

    departureTime?: string;
    arrivalTime?: string;

    inTheWayHours?: number;
    inTheWayMinutes?: number;

    distance?: number;

    orderFrom?: number;
    orderTo?: number;

    comment?: string;

    details?: RouteStopTicketDetailDto[];
}

File: Railway.API/Railway.Infrastructure/Dtos/Procedures/RouteSearch/RouteStopTicketDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Railway.Infrastructure.Dtos.Procedures.RouteSearch;

public class RouteStopTicketDto
{
    public int Id { get; set; }

    public int RouteId { get; set; }

    public string RouteFromLocality { get; set; }

    public string RouteToLocality { get; set; }

    public TimeSpan? DepartureTime { get; set; }

    public TimeSpan? ArrivalTime { get; set; }

    public int? InTheWayHours { get; set; }

    public int? InTheWayMinutes { get; set; }

    public double? Distance { get; set; }

    public int OrderFrom { get; set; }

    public int OrderTo { get; set; }

    public string Comment { get; set; }

    [NotMapped]
    public ICollection<RouteStopTicketDetailDto> Details { get; set; }
}
